cmd/activator/main.go

```diff
@@ -133,6 +152,81 @@ func main() {
 		logger.Fatalw("Failed to create stats reporter", zap.Error(err))
 	}
 
+	// Set up signals so we handle the first shutdown signal gracefully.
+	stopCh := signals.SetupSignalHandler()
+
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, defaultResyncInterval)
+	servingInformerFactory := servinginformers.NewSharedInformerFactory(servingClient, defaultResyncInterval)
+	endpointInformer := kubeInformerFactory.Core().V1().Endpoints()
+	revisionInformer := servingInformerFactory.Serving().V1alpha1().Revisions()
+
+	params := queue.BreakerParams{QueueDepth: breakerQueueDepth, MaxConcurrency: breakerMaxConcurrency, InitialCapacity: 0}
+
+	// Return the number of endpoints, 0 if no endpoints are found.
+	endpointsGetter := func(revID activator.RevisionID) (int32, error) {
+		endpoints, err := endpointInformer.Lister().Endpoints(revID.Namespace).Get(revID.Name)
+		if errors.IsNotFound(err) {
+			return 0, nil
+		}
+		if err != nil {
+			return 0, err
+		}
+		addresses := activator.EndpointsAddressCount(endpoints.Subsets)
+		return int32(addresses), nil
+	}
+
+	// Return the revision from the observer.
+	revisionGetter := func(revID activator.RevisionID) (*v1alpha1.Revision, error) {
+		return revisionInformer.Lister().Revisions(revID.Namespace).Get(revID.Name)
+	}
+
+	throttlerParams := activator.ThrottlerParams{
+		BreakerParams: params,
+		Logger:        logger,
+		GetEndpoints:  endpointsGetter,
+		GetRevision:   revisionGetter,
+	}
+	throttler := activator.NewThrottler(throttlerParams)
+
+	handler := cache.ResourceEventHandlerFuncs{
+		AddFunc:    activator.UpdateEndpoints(throttler),
+		UpdateFunc: controller.PassNew(activator.UpdateEndpoints(throttler)),
+	}
+
+	filter := func(obj interface{}) bool {
+		endpoints := obj.(*corev1.Endpoints)
+		// Pass only the endpoints created by revisions.
+		revisionID := serving.RevisionUID
+		_, ok := endpoints.Labels[revisionID]
+		return ok
+	}
```

Comment:
Can we generalize this in `pkg/reconciler/filter.go` as an `AnnotationExistsFilterFunc`? There's already a filter function there to check for specific annotations. This seems useful (and possibly allows us to inline `handler` and that function into the `AddEventHandler` call below?